pkg/services/provisioning/plugins/config_reader.go

```diff
@@ -90,18 +91,16 @@ func (cr *configReaderImpl) parsePluginConfig(path string, file fs.DirEntry) (*p
 
 func validateRequiredField(apps []*pluginsAsConfig) error {
 	for i := range apps {
-		var errStrings []string
+		errs := []error{}
 		for index, app := range apps[i].Apps {
 			if app.PluginID == "" {
-				errStrings = append(
-					errStrings,
-					fmt.Sprintf("app item %d in configuration doesn't contain required field type", index+1),
-				)
+				err := fmt.Errorf("app item %d in configuration doesn't contain required field type", index+1)
+				errs = append(errs, err)
 			}
 		}
 
-		if len(errStrings) != 0 {
-			return fmt.Errorf(strings.Join(errStrings, "\n"))
+		if len(errs) != 0 {
+			return errors.Join(errs...)
```

Comment:
This does not have the same semantic result. I assume you've verified this is fine.